docgen/src/guide/Migration_guide_package_import.md

```diff
@@ -0,0 +1,100 @@
+---
+title: Migration Guide - Package import
+mainTitle: Guides
+layout: main.pug
+category: guide
+navWeight: 10
+---
+
+With the release of the version `5.2.0` we completely revamped the architecture of the package. We tried to keep the package backward compatibility as possible. The only case **where changes are required** is when you are using the private imports. Like the following example:
+
+```js
+// Still work with `react-instantsearch` ≥ `5.2.0`
+import { InstantSearch } from 'react-instantsearch/dom';
+
+// Doesn't work with `react-instantsearch` ≥ `5.2.0`
+import { InstantSearch } from 'react-instantsearch/es/widgets/SearchBox';
+```
+
+We never documentted the latter but it was handy to use. Mainly because the default import was not compatible with [tree shaking](https://webpack.js.org/guides/tree-shaking). For this reason we decided to update our package **to be compatible with tree shaking by default**. To achieve this we choose to publish three new packages:
+
+- `react-instantsearch-core`: contains all the core logic of React InstantSearch
+- `react-instantsearch-dom`: contains the DOM specfic widgets and components
+- `react-instantsearch-native`: contains the React Native specfic widgets and components
+
+The `react-instantsearch-core` package is used internally by the two others. Most of the time you don't need to directly use this package. One use case for it usage is the creation of a new "binding". We only support [React DOM][react-website] and [React Native][react-native-website]. But the core package is not tied to any rendering engine. It means that you can create your own binding for [React 360][react-360-website] for example.
```

Comment:
"One use case for it usage" → "One use case for its usage" or "One use case for it is the creation"?